common/api/frontend-tiles.api.md

```diff
@@ -28,6 +36,36 @@ export const frontendTilesOptions: {
     useIndexedDBCache: boolean;
 };
 
+// @beta
+export interface GraphicRepresentation {
+    dataSource: DataSource;
+    displayName: string;
+    format: GraphicRepresentationFormat;
```

Comment:
tsc's exhaustiveness checks mean that you will not be able to add new values to the enum without potentially breaking somebody's code. I expect we will add new formats in the future.

Reply:
We had a long discussion about it. While we agree that we will add new formats in the future, in the short-mid term we are the ones that will add new formats/providers, so an enum is good for now. We thought about implementing a union, which could be extended, but we thought about keeping this simple for now.

If we want to add new formats, we should use a "general" provider that provides functionality to register data sources, but if we go in that direction, we should rethink also the other providers (e.g. reality data). So, I'd personally keep this simple for now, and in parallel have a serious discussion about unifying the data sources in the iTwin Viewer in global.

Reply:
> We had a long discussion about it.

Who is "we"?

> While we agree that we will add new formats in the future, in the short-mid term we are the ones that will add new formats/providers, so an enum is good for now.

When you introduce those new formats, you run the risk of introducing run-time exceptions and/or compile-time errors in other people's code.

Reply:
> Who is "we"?
Andre, Erin and me.

> When you introduce those new formats, you run the risk of introducing run-time exceptions and/or compile-time errors in other people's code.

Can you give me an example? Adding a new value to the enum shouldn't break anything, right? I guess I'm missing something here or maybe thinking "C++ minded".
The goal is to make sure that users are restricted to only use the supported formats.
The restriction can easily be removed, but isn't it better to restrict it to produce compile-time errors if the format is not well defined?

Reply:
```
enum Enum { A, B };
function getAThing(enum: Enum): Thing {
  switch (enum) {
    case Enum.A: return new ThingA();
    case Enum.B: return new ThingB();
    // no default required - every possible case covered
  }
  // no default return required - unreachable code
}
```

Now add `Enum.C` and suddenly `getAThing` can return `undefined` at run-time. If the author of the code above recompiles, compilation will fail.

Reply:
Ahhhh, thanks for sharing!
Unfortunately, that makes all the sense 😅 
Ok, then enum out.

Reply:
For documentation purposes you could use a type like `"IMDL" | "3DTILES" | string`.

Reply:
the enum has been removed and replaced with a type of `"IMDL" | "3DTILES" | string`